### DISCUSSION TITLE
How to add route-specific OpenAPI documentation?

### DISCUSSION THREAD

**Discussion by RachitBhargava99** [fastapi/fastapi#8870] [Questions] [answered]
Let's say that I want to add some extra information to the automatically generated OpenAPI documentation, but I do not want to write the entire OpenAPI file myself for the entire application (just some of the routes). How do I do that?

I am looking to add information about what is expected in the request body (I can do that from Pydantic schemas, but is there a way to add a comment about what each of the requirement is supposed to be?), what information is expected to be returned (with a short description of what is being returned) and the error codes that the route will return (along with a description of why that might happen).

```Python
from fastapi import FastAPI, HTTPException

app = FastAPI()


@app.get("/", response_model=some_pydantic_schema)
def read_root(request_model=some_other_pydantic_schema):
    if True is False:
      raise HTTPException(status_code=500, detail="Well, Python is just stupid, nothing you can do about it.")
    return {"Hello": "World"}
```

**Comment by Kludex** [comment, marked as answer]
>  what information is expected to be returned (with a short description of what is being returned) and the error codes that the route will return (along with a description of why that might happen).

https://fastapi.tiangolo.com/advanced/additional-responses

> I am looking to add information about what is expected in the request body (I can do that from Pydantic schemas, but is there a way to add a comment about what each of the requirement is supposed to be?)

```python
from pydantic import BaseModel, Field

class MyModel(BaseModel):
    a: str = Field(..., description="Yes, you can add a description")
```

**Comment by SBalaSundar** [comment]
If you want to add detailed description (a paragraph). You can add the information in the function / endpoint description.
Else you can use the solution given by @Kludex 
```
from fastapi import FastAPI, HTTPException
from pydantic import BaseModel, Field

app = FastAPI()

class RequestPydanticSchema(BaseModel):
       field1: str = Field(..., description="Add description of field1")
       field2: int = Field(..., description="Add description of field2")

@app.get("/", response_model=some_pydantic_schema)
def read_root(request_model=RequestPydanticSchema):

    """ Add your detailed information here """

    if True is False:
      raise HTTPException(status_code=500, detail="Well, Python is just stupid, nothing you can do about it.")
    return {"Hello": "World"}
```
